tests/test_mapping_filed.py

```diff
@@ -0,0 +1,118 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+import numpy as np
+import torch
+from parameterized import parameterized
+
+from monai.data import DataLoader, Dataset, decollate_batch
+from monai.inferers import sliding_window_inference
+from monai.networks.nets import UNet
+from monai.transforms import Compose, EnsureChannelFirstd, LoadImaged, SaveImaged, WriteFileMappingd
+from monai.utils import optional_import
+
+nib, has_nib = optional_import("nibabel")
+
+
+def create_input_file(temp_dir, name):
+    test_image = np.random.rand(128, 128, 128)
+    input_file = os.path.join(temp_dir, name + ".nii.gz")
+    nib.save(nib.Nifti1Image(test_image, np.eye(4)), input_file)
+    return input_file
+
+
+TEST_CASE_1 = [["seg"], ["seg"]]
+TEST_CASE_2 = [["seg"], ["image"]]
+TEST_CASE_3 = [["image"], ["seg"]]
+TEST_CASE_4 = [["image", "seg"], ["seg"]]
+TEST_CASE_5 = [["seg"], ["image", "seg"]]
+
+
+@unittest.skipUnless(has_nib, "nibabel required")
+class TestWriteFileMappingd(unittest.TestCase):
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.output_dir = os.path.join(self.temp_dir, "output")
+        os.makedirs(self.output_dir)
+        self.mapping_file_path = os.path.join(self.temp_dir, "mapping.json")
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+        if os.path.exists(self.mapping_file_path):
+            os.remove(self.mapping_file_path)
+
+    @parameterized.expand([TEST_CASE_1, TEST_CASE_2, TEST_CASE_3, TEST_CASE_4, TEST_CASE_5])
+    def test_mapping_filed(self, save_keys, write_keys):
+
+        name = "test_image"
+        input_file = create_input_file(self.temp_dir, name)
+        output_file = os.path.join(self.output_dir, name, name + "_seg.nii.gz")
+        data = [{"image": input_file}]
+
+        test_transforms = Compose([LoadImaged(keys=["image"]), EnsureChannelFirstd(keys=["image"])])
+
+        post_transforms = Compose(
+            [
+                SaveImaged(
+                    keys=save_keys,
+                    meta_keys="image_meta_dict",
+                    output_dir=self.output_dir,
+                    output_postfix="seg",
+                    savepath_in_metadict=True,
+                ),
+                WriteFileMappingd(keys=write_keys, mapping_file_path=self.mapping_file_path),
+            ]
+        )
+
+        dataset = Dataset(data=data, transform=test_transforms)
+        dataloader = DataLoader(dataset, batch_size=1)
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        model = UNet(spatial_dims=3, in_channels=1, out_channels=2, channels=(16, 32), strides=(2,)).to(device)
+        model.eval()
+
+        try:

```

Comment:
Instead of using try-except in test case, I would suggest let's clearly distinguish between cases that may cause errors and not.